lib/rubocop/cop/style/redundant_parentheses.rb

```diff
@@ -201,6 +201,12 @@ def argument_of_parenthesized_method_call?(begin_node, node)
           parent.call_type? && parent.parenthesized? && parent.receiver != begin_node
         end
 
+        def oneline_rescue_parentheses_required?(begin_node, node)
+          return false unless (parent = begin_node.parent)
+
+          !parent.type?(:call, :array, :pair) && node.rescue_type?
```

Comment:
I’m not very familiar with RuboCop’s internals, so apologies if this is obvious.

Is `node.rescue_type?` considered costly to call and/or more statistically relevant compared to `begin_node.parent`?

Would it make sense to use it as the first check in the guard clause or as a stand-alone guard clause, similar to other predicates like `keyword_with_redundant_parentheses?`

Reply:
Which one is more costly may depend on the code being analyzed, but using it as a guard condition seems like a reasonable idea.